Branch Alban FERAUD (#259)
APTITUDE-Consortium/aptitude-eudi-wallet-specs: docs/rulebook/wp3/dtc-rulebook.md

* Chapter 4, which specifies attestation usage.
* Chapter 5, which specifies trust anchors.
* Chapter 6, which specifies revocation.
* Chapter 7, which specifies compliance with ICAO specification.

> * Chapter 8, which discusses consideration of issuance of APTITUDE DTC.

Reviewer: ```suggestion
* Chapter 8, which discusses consideration for the issuance of the APTITUDE DTC.
```

The objective is to preserve a single interoperable DTC representation that is:

> * aligned with ICAO DTC Type 2 philosophy and eMRTD LDS semantics and data model,

Reviewer 2: Hi @AlbanFeraud , this line uses the term "philosophy". While the intention is understandable, this wordingis rather ambiguous for a technical specification and may leave readers unsure about what level of deviation is acceptable. Could we replace it with a clearer expression that states the APTITUDE DTC follows the ICAO Type 2 model but may diverge in the binding mechanism as long as the functional intent remains the same? One possible wording is:‘…aligned with the ICAO DTC Type 2 model while respecting the eMRTD LDS semantics. The binding between the virtual credential and the wallet component may deviate from the exact ICAO mechanism, provided that the overall functional intent defined by ICAO is preserved.’ This makes the requirement explicit and easier to interpret for implementers. Thanks for your great work contributing to our success👍

Reviewer: This is already resolved using concept instead of philosophy.

> * aligned with ICAO DTC Type 2 philosophy and eMRTD LDS semantics and data model,

Reviewer 3: ```suggestion
* aligned with ICAO DTC Type 2 concept and eMRTD LDS semantics and data model,
```

I find _philosophy_ in this context not fitting. What about _concept_?

Author: Agreed

* compatible with EUDI Wallet proximity presentation,
* suitable for both on-site border-control use cases and remote wallet-driven presentations,

> * capable of preserving the cryptographic binding between the virtual credential and the Wallet Secure Component Application (WSCA).

Reviewer 3: ```suggestion
* capable of preserving the cryptographic binding between the virtual credential and the Wallet Secure Component Device (WSCD).
```

i would stay with binding to the WSCD (which holds the private key).
This is aligned with language in [ARF](https://eudi.dev/3.0.0/discussion-topics/g-zero-knowledge-proof/?h=binding+wsc#42-device-binding-using-wscd)

WSCA is a software that enables the wallet to use the private key.

Author: Strictly speaking, WSCA would be more accurate than WSCD, as the WSCA also includes the application managing the cryptographic keys. However, if this wording is already used in the ARF, I don't mind using it here, despite it is not accurate

The objective is to preserve a single interoperable DTC representation that is:

> * aligned with ICAO DTC Type 2 philosophy and eMRTD LDS semantics and data model,

Reviewer: ```suggestion
* aligned with ICAO DTC Type 2 concept and eMRTD LDS semantics and data model,
```

| DTC_AE_01 | APTITUDE DTC SHALL use ISO/IEC 23220-4 PhotoID as the sole credential format. |
| DTC_AE_02 | APTITUDE DTC SHALL use ISO/IEC 18013-5 mdoc-cbor encoding for the PhotoID credential. |
| DTC_AE_03 | APTITUDE DTC SHALL support NFC engagement for proximity presentation and BLE data retrieval for Android and iOS. |
| DTC_AE_04 | APTITUDE DTC SHALL preserve ICAO LDS semantics and data model, including at least EF.DG1, EF.DG2, EF.DG14, EF.SOD.|
| DTC_AE_05 | APTITUDE DTC SHALL preserve the ISO/IEC 23220-4 PhotoID profile. |

> | DTC_AE_06 | APTITUDE DTC SHALL adopt open, standard-based encoding to maximize interoperability and avoid vendor lock-in. <br><br> Note : "open" means that the specification is public and free to use |

Reviewer: ```suggestion
| DTC_AE_06 | APTITUDE DTC SHALL adopt open, standard-based encoding to maximize interoperability and avoid vendor lock-in. <br><br> Note : "open" means that the specification is public and free to use. |
```

| Index | Requirement specification |
| --- | --- |

> | DTC_IS_01 | APTITUDE DTC SHALL be issued exclusively by the National Passport Issuing Authority of the Member State that issued the corresponding physical eMRTD. <br><br> Note : this requirement applies to the issuing authority and issuing subdivision used for the issuance of the APTITUDE DTC |

Reviewer: ```suggestion
| DTC_IS_01 | APTITUDE DTC SHALL be issued exclusively by the National Passport Issuing Authority of the Member State that issued the corresponding physical eMRTD. <br><br> Note : this requirement applies to the issuing authority and issuing subdivision used for the issuance of the APTITUDE DTC. |
```

| DTC_IS_02 | APTITUDE DTC SHALL be both issued (1) alongside the issuance of new eMRTDs, or (2) for already issued eMRTDs, except where the national authentic sources or issuing authorities require a restriction. |

> | DTC_IS_03 | The issuance process SHALL result in an ICAO DTC Type 2 (eMRTD-PC bound), where the virtual component is cryptographically linked to the WSCA being the physical component within the EUDI Wallet. |

Reviewer: ```suggestion
| DTC_IS_03 | The issuance process SHALL result in an ICAO DTC Type 2 (eMRTD-PC bound), where the virtual component is cryptographically linked to the WSCD being the physical component within the EUDI Wallet. |
```

| DTC_IS_04 | The system SHALL support the complete lifecycle management of the DTC, including secure revocation and update mechanisms managed by the issuing authority. |

#### Table 3 — Requirements on data elements

| Index | Requirement specification |
| --- | --- |
| DTC_DM_01 | The APTITUDE DTC SHALL contain DG1, DG2, DG14, SOD  as derived from the physical eMRTD passport and MAY contain other data groups allowed by ICAO DTC-VC specifications, as long as they are also present in the corresponding physical eMRTD. |

> | DTC_DM_02 | The APTITUDE DTC MAY contain additional attributes beyond those found in the eMRTD dataset |

Reviewer: ```suggestion
| DTC_DM_02 | The APTITUDE DTC MAY contain additional attributes beyond those available in the eMRTD dataset.|
```

| ``birth_date`` | according to [ISO/IEC 23220-2.2] | 01-01-1980 |
| ``portrait`` | according to [ISO/IEC 23220-2.2] | ... |
| ``age_over_18`` | according to [ISO/IEC 23220-2.2] | T  |

> | ``document_number`` | identifier of the APTITUDE DTC according to [ISO/IEC 23220-2.2] | YA1234567 |

Reviewer 2: In addition, ICAO has agreed that [§2.1.4 of WG3TF5_N0290 TR - Digital Travel Credentials v1.2]
> dtcSecurityInfo [22] EXPLICIT DTCSecurityInfo OPTIONAL,
>[...]  This field MUST NOT be present if DTC is eMRTD Bound

> | ``document_number`` | identifier of the APTITUDE DTC according to [ISO/IEC 23220-2.2] | YA1234567 |

Reviewer 4: will be discussed at F2F meeting August 27

| **Identifier** | **Description** | **Example** |
| --- | --- | --- |

> | ``issue_date`` | date of issuance of the APTITUDE DTC according to [ISO/IEC 23220-2.2] | 20-04-2023 |

Reviewer: I am ok with adding the description, but right now we have description for some attributes only, and the others only provide references.

| **Identifier** | **Description** | **Example** |
| --- | --- | --- |
| ``issuing_subdivision`` | according to [ISO/IEC 23220-2.2] | TN  |

> | ``administrative_number`` | according to [ISO/IEC 23220-4] | 9876543210 |

Reviewer 2: you're absolutely right, 23220-2 doesn't define these, only -4. so
approved

| ``dg11`` | ``dg11`` | O |
| ``dg12`` | ``dg12`` | O |
| ``dg13`` | ``dg13`` | O |

> | ``dg14`` | no further information<br><br> *Note:* DG14 is mandatory in EU/SAC eMRTDs | M |

Reviewer 2: Please spell out this acronym, thanks

## 5 Trust Anchors

The APTITUDE DTC is derived from data contained in the LDS data groups of the corresponding physical eMRTD and is signed by the national issuing authority. The issuing authority SHALL sign the issuer signed data, i.e. the Mobile Security Object (MSO), using a DTC signer key and certificate under the respective CSCA root certificate.
The document signer key and certificate SHALL comply with clause 2.2 in [ICAO-DTC-VC-TR] and SHALL meet the following conditions:

> * The DTC signer certificate SHALL include the following OID in the extendedKeyUsage extension : xxxxx;

Reviewer 2: I understand you're proposing for this rulebook to specify a dedicated OID value from yet to be defined usage and validity period, rather than adopting 2.23.136.1.1.12.1 in line with [ICAO-DTC-VC-TR]?
Could you share your reasons for this?

Reviewer 4: will be discussed at F2F meeting August 27

Revocation of the linked eMRTD and LDS data given in the ICAO PhotoID data elements remains unchanged and is governed by the applicable ICAO, eMRTD and national procedures.

## 7 Compliance with ICAO specifications

After successfully processing the device response and the verification procedure, a reader MAY recontruct from the content of the APTITUDE DTC Attestation an ICAO compliant DTC-VC as defined in [ICAO-DTC-VC-TR], supporting the following Type of ICAO DTC:

* ICAO DTC Type 1, also named eMRTD bound DTC;

> * eMRTD bound extended DTC;

Reviewer 3: What is the motivation for having this 'extendend DTC'?

It's neither ICAO Type 1 nor Type 2?

If there's a good reason to have this intermediate state, it should be well-explained here.

Author: I propose to also envision this new type of DTC, as it is about to be reintroduced by ICAO in the next update of the ICAO TR. This type of DTC is still bound to an eMRTD only, but contains supplemental data alongside the LDS. This is the reason why the field DTCSignerInfo is required, to seal alltogether all the data.

Reviewer 3: Ok, thanks for your clarification. I was not aware of this new type. And I guess that the motivation for this new ICAO type is that a specific DTC-VC can be revoked without revoking the eMRTD itself? Or is there another reason?

Would that mean, that an DTC Aptitude issuer which issues eMRTD bound extended DTC would (in worst case) have to maintain two places with revocation information. One that is compliant to mdoc (chapter 6.1) and one that holds a list of revoked DTCIdentifier?

Can a MSO revocation list (18013-5-2 ch 12.3.6.3) could do the job for both worlds?

Author: Hello
1/As per the latest draft of the TR, the eMRTD bound extended DTC would bear the same identifeir as the underlying eMRTD. Yet this may change during the next round of techncial discussion on that matter
2/I think the revocation would work as for the DTC type 1 : revocation of the MSO and update of the SLTD.
I hope it clarifies.

Reviewer 5: Hello,

in my honest opinion, using the same identifier sounds strange. Because the idea of having the identifier was to use it as a way to revoke the DTC VC independent of eMRTD. Otherwise, what would be the added value here?

Author: It is a feedback I can carry out to ICAO so that that type of DTC has an independant lifecycle

Reviewer 3: I agree. If there is no extra identifier, I see no benefit.
If there is really no benefit for us, we should not add it to Aptitude DTC.

Reviewer 2: Per minutes from July 30th, " consensus was that the current rulebook should remain focused on its core objectives. As a result, the preferred direction is to include only a high-level reference to this option, together with references to relevant ICAO work, rather than developing a full specification at this stage."
@AlbanFeraud  I therefore take it you go back to the version without "eMRTD bound extended DTC" ?

**ICAO DTC Type 1, also named eMRTD bound DTC**
In this case the reader SHALL bundle the ``dg1``, ``dg2``, ``sod``, and if present ``dg3``, ``dg4``, ``dg5``, ``dg6``, ``dg7``, ``dg8``, ``dg9``, ``dg10``, ``dg11``, ``dg12``, ``dg13``, ``dg14``, ``dg15``, ``dg16`` within the structure ``DTCData`` to build the structure ``DTCContentInfo``.

*Note:* In this case, the structure ``DTCContentInfo`` does not contain stuctures ``DTCTBS``, ``DTCSignerInfo``, ``DTCSecurityInfo`` and ``DTCOtherInfo``.

**eMRTD bound extended DTC**

> In this case the reader SHALL bundle the ``dg1``, ``dg2``, ``sod``, and if present ``dg3``, ``dg4``, ``dg5``, ``dg6``, ``dg7``, ``dg8``, ``dg9``, ``dg10``, ``dg11``, ``dg12``, ``dg13``, ``dg14``, ``dg15``, ``dg16`` to get the structure ``DTCData``. Subsequently, the reader SHALL (1) append the stucture ``DTCOtherInfo`` found within the APTITUDE DTC Attestation in the structure ``DTCData`` and (2) bundle the structures ``DTCTBS`` and ``DTCSignerInfo`` found within the APTITUDE DTC Attestation together with ``DTCData`` to build the structure ``DTCContentInfo``.

Reviewer 3: > ``DTCTBS`` and ``DTCSignerInfo`` found within the APTITUDE DTC Attestation 

In my understanding there is no ``DTCSignerInfo`` encoded in the APTITUDE DTC (chapter 3?)

Author: see comment above. I still have to include these new field in the doctype (still to be done)

Reviewer 2: Per minutes from July 30th, " consensus was that the current rulebook should remain focused on its core objectives. As a result, the preferred direction is to include only a high-level reference to this option, together with references to relevant ICAO work, rather than developing a full specification at this stage."
@AlbanFeraud  I therefore take it you can remove this passage?

Reviewer 4: will be discussed at F2F meeting August 28

*Note:* In this case, the structure ``DTCContentInfo`` does not contain stuctures ``DTCTBS``, ``DTCSignerInfo``, ``DTCSecurityInfo`` and ``DTCOtherInfo``.

**eMRTD bound extended DTC**
In this case the reader SHALL bundle the ``dg1``, ``dg2``, ``sod``, and if present ``dg3``, ``dg4``, ``dg5``, ``dg6``, ``dg7``, ``dg8``, ``dg9``, ``dg10``, ``dg11``, ``dg12``, ``dg13``, ``dg14``, ``dg15``, ``dg16`` to get the structure ``DTCData``. Subsequently, the reader SHALL (1) append the stucture ``DTCOtherInfo`` found within the APTITUDE DTC Attestation in the structure ``DTCData`` and (2) bundle the structures ``DTCTBS`` and ``DTCSignerInfo`` found within the APTITUDE DTC Attestation together with ``DTCData`` to build the structure ``DTCContentInfo``.
Conversly, during issuance of the APTITUDE DTC, the issuing authority SHALL compute the following structures in accordance with [ICAO-DTC-VC-TR] and store them within the APTITUDE DTC Attestation:

* ``DTCTBS``
* ``DTCOtherInfo``
* ``DTCSignerInfo``

*Note:* In this case, the structure ``DTCData`` does not contain the stucture ``DTCSecurityInfo``.

> The binding between the DTC-VC and the EUDI Wallet enabled by the Rulebook is not compliant with the mechanisms currently defined by ICAO for the binding between the DTC-VC and DTC-PC. Therefore, the following features allowed by the Rulebook are currently not compliant with ICAO specifications:

Reviewer 2: As per minutes from July 30th, please "adopt a positive and factual tone, describing the adopted approach and its characteristics rather than framing the discussion solely in terms of compliance or non-compliance" thanks :)

Reviewer 4: will be discussed at F2F meeting August 27

}
```

## 8 Consideration for issuance of APTITUDE DTC credentials

> APTITUDE DTC is bound to an existing eMRTD, either issued before the APTITUDE DTC (preexisting eMRTD) or issued simultaneously.

Reviewer: ```suggestion
APTITUDE DTC is bound to an existing eMRTD, either issued before the APTITUDE DTC (pre-existing eMRTD) or issued simultaneously.
```
